tools/autograd/derivatives.yaml

```diff
@@ -930,6 +930,9 @@
 - name: repeat(Tensor self, int[] repeats) -> Tensor
   self: repeat_backward(grad, repeats, self.sizes())
 
+- name: special_entr(Tensor self) -> Tensor
+  self: grad * (-(1 + self.log()))
```

Comment:
Would you explain the derivative of this function?

The function is tripartite, and so is not technically differentiable. Everywhere self is negative it seems like the derivative would be zero? Then at zero we can probably assign whatever derivative we like, but zero seems appropriate.

Now let's talk about the derivative of the operation where x > 0. Wouldn't it be -log(x) - 1/(ln 2)? Because the logarithm computed here is in base 2 and not the natural logarithm?

Reply:
For negative values, derivative would indeed be `nan`, 
```python
>>> t = torch.tensor(-1.)
>>> -1 - t.log()
tensor(nan)
```

On log being base `2`, it is actually natural log in scipy ( i think base 2 is used in context of information theory).

```python
>>> import scipy.special
>>> import math
>>> scipy.special.entr(2)
-1.3862943611198906
>>> -2 * math.log2(2)
-2.0
>>> -2 * math.log(2)
-1.3862943611198906
```

Reply:
I always forget that std::log is base e, thanks for correcting me. Let's update the documentation with `\ln` then to be clear about the base?